src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,39 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + (width != null ? width.hashCode() : 0);
+        result = 31 * result + (length != null ? length.hashCode() : 0);
+        result = 31 * result + (color != null ? color.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || !obj.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        try {
+            Rectangle rectangle = (Rectangle) obj;
+            return (width == null || rectangle.width == null) ? width == rectangle.width
+                    : width.equals(rectangle.width)
+                    && (length == null || rectangle.length == null) ? length == rectangle.length
+                    : length.equals(rectangle.length)
+                    && (color == null || rectangle.color == null) ? color == rectangle.color
+                    : color.equals(rectangle.color);
+        } catch (ClassCastException e) {
+            e.printStackTrace();
+        }
```

Comment:
1)We don't catch RuntimeExceptions(Which ClassCastException is). Mostly the occur due to a programmer mistake in code. 
2) Don't ignore exceptions if you are catching them(rethrow them);
3) !obj.getClass().equals(Rectangle.class) 
this check guarantees that such exception will never occur&